### ISSUE TITLE
SmartCity-Accessibility Server Side Unit Tests #923 failed

### ISSUE THREAD

**Issue by JuanJenkinsBot** [TechnionYP5777/SmartCity-Accessibility#443]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 7.81 KB...]
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in smartcity.accessibility.socialnetwork.HelpfulnessTest
Running smartcity.accessibility.socialnetwork.UserFunctionalityTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.024 sec - in smartcity.accessibility.socialnetwork.UserFunctionalityTest
Running smartcity.accessibility.socialnetwork.BestReviewsTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec - in smartcity.accessibility.socialnetwork.BestReviewsTest
Running smartcity.accessibility.socialnetwork.UserPinTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.03 sec - in smartcity.accessibility.socialnetwork.UserPinTest
Running smartcity.accessibility.socialnetwork.DeleteReviewsTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.085 sec - in smartcity.accessibility.socialnetwork.DeleteReviewsTest
Running smartcity.accessibility.socialnetwork.ScoreTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.663 sec - in smartcity.accessibility.socialnetwork.ScoreTest
Running smartcity.accessibility.database.UserManagerTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.44 sec - in smartcity.accessibility.database.UserManagerTest
Running smartcity.accessibility.database.LocationManagerTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.841 sec - in smartcity.accessibility.database.LocationManagerTest
Running smartcity.accessibility.database.ReviewManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.561 sec - in smartcity.accessibility.database.ReviewManagerTest
Running smartcity.accessibility.database.UserProfileManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.263 sec - in smartcity.accessibility.database.UserProfileManagerTest
Running smartcity.accessibility.search.SearchTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.253 sec - in smartcity.accessibility.search.SearchTest

Results :

Failed tests: 
  ReviewCommentTest.c:61 expected:<0> but was:<2>
  ReviewCommentTest.d:70 expected:<-1> but was:<0>
  ReviewCommentTest.e:80 expected:<6> but was:<129>

Tests run: 80, Failures: 3, Errors: 0, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:16 min
[INFO] Finished at: 2017-06-18T12:55:15+00:00
[INFO] Final Memory: 34M/212M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: There are test failures.
[ERROR] 
[ERROR] Please refer to /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [Arthur Sapozhnikov] 6ce8f3f707e5fb28631618ebe27e7f50ec2c5fad - #442 it seems that ReviewComment class didn't even have the option to

[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/923/)

**Comment by JuanJenkinsBot** [comment]
Build 'SmartCity-Accessibility Server Side Unit Tests' is failing!

Last 50 lines of build :
```
[...truncated 7.81 KB...]
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 sec - in smartcity.accessibility.socialnetwork.HelpfulnessTest
Running smartcity.accessibility.socialnetwork.UserFunctionalityTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.026 sec - in smartcity.accessibility.socialnetwork.UserFunctionalityTest
Running smartcity.accessibility.socialnetwork.BestReviewsTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.008 sec - in smartcity.accessibility.socialnetwork.BestReviewsTest
Running smartcity.accessibility.socialnetwork.UserPinTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.019 sec - in smartcity.accessibility.socialnetwork.UserPinTest
Running smartcity.accessibility.socialnetwork.DeleteReviewsTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.039 sec - in smartcity.accessibility.socialnetwork.DeleteReviewsTest
Running smartcity.accessibility.socialnetwork.ScoreTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.725 sec - in smartcity.accessibility.socialnetwork.ScoreTest
Running smartcity.accessibility.database.UserManagerTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.397 sec - in smartcity.accessibility.database.UserManagerTest
Running smartcity.accessibility.database.LocationManagerTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.107 sec - in smartcity.accessibility.database.LocationManagerTest
Running smartcity.accessibility.database.ReviewManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.704 sec - in smartcity.accessibility.database.ReviewManagerTest
Running smartcity.accessibility.database.UserProfileManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.317 sec - in smartcity.accessibility.database.UserProfileManagerTest
Running smartcity.accessibility.search.SearchTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.148 sec - in smartcity.accessibility.search.SearchTest

Results :

Failed tests: 
  ReviewCommentTest.c:61 expected:<0> but was:<2>
  ReviewCommentTest.d:70 expected:<-1> but was:<0>
  ReviewCommentTest.e:80 expected:<6> but was:<129>

Tests run: 80, Failures: 3, Errors: 0, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:16 min
[INFO] Finished at: 2017-06-18T13:19:36+00:00
[INFO] Final Memory: 35M/205M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project SmartCity-Accessibility: There are test failures.
[ERROR] 
[ERROR] Please refer to /var/lib/jenkins/sharedspace/SmartCity-Accessibility/server-side/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
```

Changes since last successful build:
- [Arthur Sapozhnikov] 6ce8f3f707e5fb28631618ebe27e7f50ec2c5fad - #442 it seems that ReviewComment class didn't even have the option to

- [Arthur Sapozhnikov] 59c61eea4e5f9eba43fee33868f0f8c5203d9c80 - #443 fixed equals() function and refactored hashCode() function.



[View full output](http://kapalex.ddns.net:8080/job/SmartCity-Accessibility%20Server%20Side%20Unit%20Tests/924/)

**Comment by JuanJenkinsBot** [comment]
Build was fixed!
